File: src/main/java/com/wgzhao/addax/admin/controller/AuthController.java
package com.wgzhao.addax.admin.controller;


import com.wgzhao.addax.admin.dto.ApiResponse;
import com.wgzhao.addax.admin.dto.AuthRequestDTO;
import com.wgzhao.addax.admin.service.JwtService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@CrossOrigin
@RequestMapping("/auth")
@RequiredArgsConstructor
@Slf4j
public class AuthController {

    private final JwtService jwtService;
    private final AuthenticationManager authenticationManager;


    @PostMapping("/login")
    public ApiResponse<String> AuthenticateAndGetToken(@RequestBody AuthRequestDTO authRequestDTO){
        Authentication authentication = authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(authRequestDTO.getUsername(), authRequestDTO.getPassword()));
        if(authentication.isAuthenticated()){
            return ApiResponse.success(jwtService.generateToken(authRequestDTO.getUsername()));

        } else {
            return ApiResponse.error(401, "failed to authenticate user");
        }

    }
}

File: src/main/kotlin/com/wgzhao/addax/admin/controller/AuthController.kt
package com.wgzhao.addax.admin.controller

import com.wgzhao.addax.admin.dto.ApiResponse
import com.wgzhao.addax.admin.dto.AuthRequestDTO
import com.wgzhao.addax.admin.service.JwtService
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.web.bind.annotation.*
import io.github.oshai.kotlinlogging.KotlinLogging

/**
 * 用户认证接口，提供登录认证功能
 */
@RestController
@CrossOrigin
@RequestMapping("/auth")
class AuthController(
    private val jwtService: JwtService,
    private val authenticationManager: AuthenticationManager
) {
    private val log = KotlinLogging.logger {}

    /**
     * 用户登录认证，返回JWT令牌
     * @param authRequestDTO 登录请求参数
     * @return JWT令牌或认证失败信息
     */
    @PostMapping("/login")
    fun authenticateAndGetToken(@RequestBody authRequestDTO: AuthRequestDTO): ApiResponse<String> {
        val authentication = authenticationManager.authenticate(
            UsernamePasswordAuthenticationToken(authRequestDTO.username, authRequestDTO.password)
        )
        return if (authentication.isAuthenticated) {
            ApiResponse.success(jwtService.generateToken(authRequestDTO.username))
        } else {
            ApiResponse.error(401, "failed to authenticate user")
        }
    }
}
